From a72e0e3639ebae889c8c03d9379159ec5e3b57d1 Mon Sep 17 00:00:00 2001
From: Hiep Pham <hiep.pham.zy@renesas.com>
Date: Tue, 2 Feb 2021 17:56:48 +0700
Subject: [PATCH 046/135] arm64: dts: renesas: rzg2-smarc: Enable CAN in
 devicetree.

This patch enable both 2 channels can0 and can1.
It also add aliases for its.

Signed-off-by: Hiep Pham <hiep.pham.zy@renesas.com>
---
 arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi | 40 +++++++++++++++++++++++++++++
 1 file changed, 40 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi b/arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi
index f9c872f..2913e4d 100644
--- a/arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi
+++ b/arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi
@@ -11,6 +11,8 @@
 / {
 	aliases {
 		serial0 = &scif0;
+		can0 = &can;
+		can1 = &can;
 	};
 
 	chosen {
@@ -119,6 +121,30 @@
 		output-high;
 		line-name = "sd1_pwr_en";
 	};
+
+	can0_stb {
+		gpio-hog;
+		gpios = <RZG2L_GPIO(42, 2) GPIO_ACTIVE_LOW>;
+		output-high;
+		line-name = "can0_stb";
+	};
+
+	can1_stb {
+		gpio-hog;
+		gpios = <RZG2L_GPIO(42, 3) GPIO_ACTIVE_LOW>;
+		output-high;
+		line-name = "can1_stb";
+	};
+
+	can0_pins: can0 {
+		groups = "can0_data_a";
+		function = "can0";
+	};
+
+	can1_pins: can1 {
+		groups = "can1_data_a";
+		function = "can1";
+	};
 };
 
 &scif0 {
@@ -154,3 +180,17 @@
 	sd-uhs-sdr104;
 	status = "okay";
 };
+
+&can {
+	pinctrl-0 = <&can0_pins &can1_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+	channel0 {
+		renesas,channel = <0>;
+		status = "okay";
+	};
+	channel1 {
+		renesas,channel = <1>;
+		status = "okay";
+	};
+};
-- 
2.7.4

